### PR TITLE
Azure: Unify credentials in frontend 

### PR DESCRIPTION
**Pull request by yjsong11**
The goal of this PR is to unify the Azure credentials logic in Azure Monitor by using the [shared library](https://github.com/grafana/grafana-azure-sdk-react/tree/main) when possible.

The following pattern for get/update credentials is used here and planned to be used for the rest of the datasources.
- `getCredentials()`: first call `getDatasourceCredentials()` from the shared library to get any existing non-legacy credentials. If none exist, check if legacy credentials exist (`getLegacyCredentials()`) if it is applicable for the datasource. Finally, fallback to`getDefaultCredentials()` which all datasources already implement.
- `updateCredentials()`:  call `updateDatasouceCredentials()` from the shared library which will update the common shared Azure credentials. If there are additional datasource-specific properties that need to be updated, update those afterward.

### CODE REVIEW

**Comment by yjsong11** [comment]
@aangelisc Lint issues should be fixed now

**Review by aangelisc — COMMENTED** [review]
Thank you @yjsong11! This looks great. There's just one thing that we need to rectify and then this should be fine to merge.

File: public/app/plugins/datasource/azuremonitor/credentials.ts [hunk 1]
```diff
@@ -1,259 +1,87 @@
...
-        credentials.clientId &&
-        // When ignoreSecret is set we consider the credentials complete without checking the secret
-        !!(ignoreSecret || credentials.clientSecret)
-      );
+export function updateCredentials(
+  options: AzureMonitorDataSourceSettings,
+  credentials: AzureCredentials
+): AzureMonitorDataSourceSettings {
+  options = updateDatasourceCredentials(options, credentials);
+  if (instanceOfAzureCredential<AadCurrentUserCredentials>('currentuser', credentials)) {
```
[the remark this answers is not in the document]
Comment (yjsong11) [reply public/app/plugins/datasource/azuremonitor/credentials.ts:33]: Thanks @aangelisc! I will wait for that change and remove it here.

[the remark this answers is not in the document]
Comment (aangelisc) [reply public/app/plugins/datasource/azuremonitor/credentials.ts]: Hey @yjsong11, a new release of the SDK has been cut, are you able to update and remove this line? 😊

[the remark this answers is not in the document]
Comment (aangelisc) [reply public/app/plugins/datasource/azuremonitor/credentials.ts]: Ignore me!

**Review by aangelisc — APPROVED** [review]

**Comment by aangelisc** [comment]
hey @yjsong11, would you be able to merge latest main into this branch please, we have a failing check 😞 We also have a new type assertion in `MonitorConfig.tsx`. Is there any way we can avoid doing this:

```
updateDatasourceCredentials(
        { ...options, jsonData: { ...options.jsonData, subscriptionId } } as AzureMonitorDataSourceSettings,
        credentials
      )
```

**Comment by yjsong11** [comment]
Thanks @aangelisc! Merged main and removed the type assertion.

**Pull request merged**